docs/resources/softlayer_subnet.md

```diff
@@ -0,0 +1,109 @@
+#### `softlayer_subnet`
+
+`softlayer_subnet` provides portable and static subnets that consist of either IPv4 and IPv6 addresses. Users are able to create 
+public portable subnets, private portable subnets, and public static subnets with an IPv4 option and public portable subnets and public static 
+subnets with an IPv6 option. 
+ 
+The portable IPv4 subnet is created as a seconday subnet on a VLAN. IP addresses in the portable subnet can be assigned as secondary IP 
+ addresses for SoftLayer resources in the VLAN. Each portable subnet has a default gateway IP address, network IP address, and broadcast 
+ IP address. For example, if a portable subnet is `10.0.0.0/30`, `10.0.0.0` is a network IP address, `10.0.0.1` is a default gateway IP address, 
+ and `10.0.0.3` is a broadcast IP address. Therefore, only `10.0.0.2` can be assigned to SoftLayer resources as a secondary IP address. 
+ Number of usuable IP addresses is `capacity` - 3. If `capacity` is 4, the number of usuable IP addresses is 4 - 3 = 1. If `capacity` is 8, the 
+ number of usuable IP addresses is 8 - 3 = 5. For additional details, refer to [Static and Portable IP blocks](https://knowledgelayer.softlayer.com/articles/static-and-portable-ip-blocks).
+
+The static IPv4 subnet provides secondary IP addresses for primary IP addresses. It provides secondary IP addresses for SoftLayer resources such as 
+virtual servers, bare metal servers, and netscaler VPXs. Suppose that a virtual server requires secondary IP addresses. Then, users can create 
+a static subnet on the public IP address of the virtual server. Unlike the portable subnet, `capacity` is same with a number of usuable IP address. 
+For example, if a static subnet is `10.0.0.0/30`, `capacity` is 4 and four IP addresses(10.0.0.0 ~ 10.0.0.3) can be used as secondary IP addresses. 
+For additional details, refer to [Subnet](https://knowledgelayer.softlayer.com/topic/subnets).
+
+Both the public portable IPv6 subnet and the public static IP only accept `64` as a value of `capacity` attribute. They provide 2^64 IP addresses. For additional detail, refer to [IPv6 address](http://blog.softlayer.com/tag/ipv6)
+
+The following example will create a private portable subnet which has one available IPv4 address. 
+##### Example Usage of portable subnet
+
+```hcl
+# Create a new portable subnet
+resource "softlayer_subnet" "portable_subnet" {
+  type = "Portable"
```

Comment:
if there can only be two types, portable or static, make it a boolean then `portable = true` or `static = true` to avoid possible spelling errors. If it is not portable, then it is static and vice-versa.